test/extended/olm/olmv1.go

```diff
@@ -419,3 +500,89 @@ func checkFeatureCapability(ctx context.Context, oc *exutil.CLI) {
 		g.Skip("Test only runs with OperatorLifecycleManagerV1 capability")
 	}
 }
+
+// verifyAPIEndpoint runs a job to validate the given service endpoint of a ClusterCatalog
+func verifyAPIEndpoint(ctx g.SpecContext, oc *exutil.CLI, namespace, catalogName, endpoint string) {
+	jobName := fmt.Sprintf("test-catalog-%s-%s-%s", catalogName, endpoint, rand.String(5))
+
+	baseURL, err := oc.AsAdmin().Run("get").Args(
+		"clustercatalogs.olm.operatorframework.io",
+		catalogName,
+		"-o=jsonpath={.status.urls.base}").Output()
+	o.Expect(err).NotTo(o.HaveOccurred())
+	o.Expect(baseURL).NotTo(o.BeEmpty(), fmt.Sprintf("Base URL not found for catalog %s", catalogName))
+
+	serviceURL := fmt.Sprintf("%s/api/v1/%s", baseURL, endpoint)
+	g.GinkgoLogr.Info(fmt.Sprintf("Using service URL: %s", serviceURL))
+
+	job := &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      jobName,
+			Namespace: namespace,
+		},
+		Spec: batchv1.JobSpec{
+			Template: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  "api-tester",
+							Image: "registry.redhat.io/rhel8/httpd-24:latest",
+							Command: []string{
+								"/bin/bash",
+								"-c",
+								fmt.Sprintf(`
+set -ex
+response=$(curl -s -k "%s" || echo "ERROR: Failed to access endpoint")
+if [[ "$response" == ERROR* ]]; then
+  echo "$response"
+  exit 1
+fi
+echo "$response" > /tmp/api-response
+
+# check if response can be parsed as new line delimited JSON
+if cat /tmp/api-response | jq -s . > /dev/null 2>&1; then
+  echo "Valid JSON response detected"
+  exit 0
+fi
+echo "ERROR: Invalid JSON response"
+exit 1
```

Comment:
This makes just a single request per pod, which probably isn't quite as efficient as if we could use an http load testing program to make a bunch of concurrent requests from a single process.

Not sure where to find an image with an http load tester available to the origin suite, but if we can find one, it would likely provide much more useful information (e.g. histogram of durations of the requests)